references/cli/def.go

```diff
@@ -303,7 +303,7 @@ func generateTerraformTypedComponentDefinition(cmd *cobra.Command, name, kind, p
 	}
 
 	switch provider {
-	case "aws", "azure", "alibaba", "tencent", "gcp", "baidu", "elastic", "ucloud":
+	case "aws", "azure", "alibaba", "tencent", "gcp", "baidu", "elastic", "ucloud", "vsphere":
```

Comment:
Provide a proposal to get the supported types dynamically:

The terraform controller writes a secret and CLI reads the  supported provider type list from this secret